api/base/views.py

```diff
@@ -1,24 +1,27 @@
-from rest_framework.decorators import api_view
+from rest_framework.views import APIView
 from rest_framework.response import Response
 
+
 from .utils import absolute_reverse
 from api.users.serializers import UserSerializer
 
-@api_view(('GET',))
-def root(request, format=None):
-    if request.user and not request.user.is_anonymous():
-        user = request.user
-        current_user = UserSerializer(user).data
-    else:
-        current_user = None
-    return Response({
-        'meta': {
-            'message': 'Welcome to the OSF API.',
-            'version': request.version,
-            'current_user': current_user,
-        },
-        'links': {
-            'nodes': absolute_reverse('nodes:node-list'),
-            'users': absolute_reverse('users:user-list'),
-        }
-    })
+
+class Root(APIView):
```

Comment:
See if putting the info from the settings.defaults.swagger_settings.description as a docstring here will cause that to show in the Django docs.